### PR TITLE
Fixing a mistake made with PR #24397 + fixing setting description for clarity

### PR DESCRIPTION
**Pull request by AndySDH**
Pull Request for Issue #24754 

### Summary of Changes

PR #24397, introduced in Joomla 3.9.5, added a check to display the custom field on the frontend based on a parameter that was meant to be a check for the **edit form**, and not for the frontend display.

The author of that PR said that to un-do it I should create another PR myself so here it is.

As explained in this comment https://github.com/joomla/joomla-cms/issues/24754#issuecomment-496509135, there already is another setting that controls whether a custom field should be displayed in the frontend or not.

That "Show on" setting is under "Form Options", and was/has always been for controlling where the field can be edited, not displayed. It just had a bad label/description.

With this PR I also take care of improving the description and label of that setting so that it causes no confusion on what it's actually doing and supposed to do.

### Testing Instructions

- Create a Custom Field
- Set the "Editable In" (or without this PR: "Show on") parameter to "Administrator"
- Assign a value to it in an article

### Current 3.9.6 result

- The field is not displayed on the frontend article either

### Expected result with this PR

- The field should be displayed on the frontend article, but not on the frontend editing form.

### Documentation Changes Required
Yes.
The label and description of the setting should be changed accordingly also here:
https://docs.joomla.org/J3.x:Adding_custom_fields/Parameters_for_all_Custom_Fields

### CODE REVIEW

**Comment by richard67** [comment]
@joomlahenk Please test

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 7e7571c07fa4a830cf3383036bcd211d20daff83<br /><br />I've tested it with an extended test compared to the testing instructions in the description:
- Created 2 fields, one with "Can be edited in" (or "Showon" without this PR) set to "Administrator" and the other set to "Site".
- Checked that when editing the article with both fields having assigned a value, the 1st field is editable in tab "Fields" in the artricle edit view in the backend and the 2nd field is not, and for frontend editing it is vice versa.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25062">issues.joomla.org/tracker/joomla-cms/25062</a>.</sub>

File: administrator/language/en-GB/en-GB.com_fields.ini [hunk 1]
```diff
@@ -52,8 +52,8 @@ COM_FIELDS_FIELD_REQUIRED_DESC="Is this a mandatory field?"
 COM_FIELDS_FIELD_REQUIRED_LABEL="Required"
 COM_FIELDS_FIELD_SHOW_ON_ADMIN="Administrator"
 COM_FIELDS_FIELD_SHOW_ON_BOTH="Both"
-COM_FIELDS_FIELD_SHOW_ON_DESC="On which part of the site should the field be shown?"
-COM_FIELDS_FIELD_SHOW_ON_LABEL="Show On"
+COM_FIELDS_FIELD_SHOW_ON_DESC="On which part of the site should the field be editable?"
+COM_FIELDS_FIELD_SHOW_ON_LABEL="Can be edited in"
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.com_fields.ini]: Capitalize first letter of each word since it is a label. Would this be better `Editable In`?

Comment (richard67) [reply administrator/language/en-GB/en-GB.com_fields.ini]: `Editable In` looks good to me, too.

Comment (AndySDH) [reply administrator/language/en-GB/en-GB.com_fields.ini]: Sure why not.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on a4947b412633b10d65d469bff8c81d000d9782a5<br /><br />I've tested it with an extended test compared to the testing instructions in the description:
- Created 2 fields, one with "Can be edited in" (or "Showon" without this PR) set to "Administrator" and the other set to "Site".
- Checked that when editing the article with both fields having assigned a value, the 1st field is editable in tab "Fields" in the artricle edit view in the backend and the 2nd field is not, and for frontend editing it is vice versa.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25062">issues.joomla.org/tracker/joomla-cms/25062</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on a4947b412633b10d65d469bff8c81d000d9782a5<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25062">issues.joomla.org/tracker/joomla-cms/25062</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25062">issues.joomla.org/tracker/joomla-cms/25062</a>.</sub>

**Comment by AndySDH** [comment]
By the way, I noticed that the documentation here https://docs.joomla.org/J3.x:Adding_custom_fields/Parameters_for_all_Custom_Fields is outdated beyond this PR.
Joomla 3.9 changed the way Custom Field options are arranged and it was not reflected in the documentation there

File: administrator/language/en-GB/en-GB.com_fields.ini [hunk 2]
```diff
@@ -52,8 +52,8 @@ COM_FIELDS_FIELD_REQUIRED_DESC="Is this a mandatory field?"
 COM_FIELDS_FIELD_REQUIRED_LABEL="Required"
 COM_FIELDS_FIELD_SHOW_ON_ADMIN="Administrator"
 COM_FIELDS_FIELD_SHOW_ON_BOTH="Both"
-COM_FIELDS_FIELD_SHOW_ON_DESC="On which part of the site should the field be shown?"
-COM_FIELDS_FIELD_SHOW_ON_LABEL="Show On"
+COM_FIELDS_FIELD_SHOW_ON_DESC="On which part of the site should the field be editable?"
```
Comment (alikon) [inline administrator/language/en-GB/en-GB.com_fields.ini]: `COM_FIELDS_FIELD_SHOW_ON_LABEL` and `COM_FIELDS_FIELD_SHOW_ON_DESC`  should be changed as well to reflect the change

**Review by alikon — CHANGES_REQUESTED** [review]

File: administrator/language/en-GB/en-GB.com_fields.ini (hunk 2 above)
Comment (AndySDH) [reply administrator/language/en-GB/en-GB.com_fields.ini]: What you mean?

Comment (alikon) [reply administrator/language/en-GB/en-GB.com_fields.ini]: the "show" is different from "edit" in my humble english knowledge

we have a language key string `(COM_FIELDS_FIELD_SHOW_ON_LABEL)` that not reflect the translation "Editable In"

Comment (AndySDH) [reply administrator/language/en-GB/en-GB.com_fields.ini]: Oh you mean the key string itself. No big deal but ok I will fix that too then.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 6e5a160c6de6d1f603d6c478928527e61e513adf<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25062">issues.joomla.org/tracker/joomla-cms/25062</a>.</sub>

**Comment by HLeithner** [comment]
@brianteeman would you like to check the language strings before I merge this PR?

**Comment by HLeithner** [comment]
thx

**Comment by AndySDH** [comment]
@HLeithner Was this somehow undone by recent changes? This seems to have reverted to "Show On" currently in Joomla! 3.9.10.

image

**Comment by AndySDH** [comment]
Nevermind, ignore, my bad

**Pull request merged**